### PR TITLE
feat(source): add scylladb connection

### PR DESCRIPTION
**Pull request by j3yzz**

### CODE REVIEW

**Comment by taha-ahmadi** [comment]
@gohossein @AbtinSetyani please review this merge request

**Comment by j3yzz** [comment]
> @gohossein @AbtinSetyani please review this merge request

Before the PR is merged, it is necessary for this folder structure to undergo a change. I will make these changes as soon as possible.

**Review by taha-ahmadi — APPROVED** [review]

**Comment by taha-ahmadi** [comment]
@j3yzz please fix conflicts

File: source/adapter/scylladb/initializedb/db.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	cluster.Consistency = conn.consistency
+	cluster.Keyspace = conn.keyspace
+	cluster.Timeout = timeoutCluster
+	cluster.RetryPolicy = &gocql.ExponentialBackoffRetryPolicy{
+		NumRetries: numRetries,
+		Min:        minRetryDelay,
+		Max:        maxRetryDelay,
+	}
+	cluster.PoolConfig.HostSelectionPolicy = gocql.TokenAwareHostPolicy(gocql.RoundRobinHostPolicy())
+	log.Println("cluster was created.")
```
Comment (gocastsian) [inline source/adapter/scylladb/initializedb/db.go:51]: Please remove redundant logs or at least use slog.Debug (the logger package in ormus codebase).

File: source/adapter/scylladb/initializedb/db.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+	return cluster
+}
+
+func (conn *ScyllaDBConnection) createKeyspace(session scylladb.SessionxInterface, keyspace string) error {
```
Comment (gocastsian) [inline source/adapter/scylladb/initializedb/db.go:56]: Remove `createKeyspace` method.

File: source/adapter/scylladb/initializedb/db.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import (
+	"fmt"
+	"github.com/gocql/gocql"
+	"github.com/ormushq/ormus/source/adapter/scylladb"
+	"log"
+	"time"
+)
+
+type ScyllaDBConnection struct {
+	consistency gocql.Consistency
```
Comment (gocastsian) [inline source/adapter/scylladb/initializedb/db.go:17]: Move this fields to config struct

File: source/repository/scylladb/migrate/migration.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+func New(dir, host, keyspace string) *Manager {
+	return &Manager{
+		dir:      dir,
+		host:     host,
+		keyspace: keyspace,
+	}
+}
+
+func (m *Manager) Run() error {
+	sourceDriver, err := iofs.New(os.DirFS(m.dir), "migrations")
```
Comment (gocastsian) [inline source/repository/scylladb/migrate/migration.go:33]: If we can `embed` migration folder to binary please add `embed` statement to migration code.

File: source/adapter/scylladb/initializedb/db.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
+/*
+Package initializedb provides functionality for initializing a connection to ScyllaDB database using to gocql library.
+
+Note: Make sure to handle errors appropriately when using this package.
+*/
+package initializedb
```
[the remark this answers is not in the document]
Comment (j3yzz) [reply source/adapter/scylladb/initializedb/db.go:6]: The things you mentioned were added in this PR.
https://github.com/ormushq/ormus/pull/42

**Pull request merged**